### PR TITLE
docs(terraform): rewrite readme

### LINKED ISSUE #17975: Rewrite Terraform manager readme
### Describe the proposed change(s).

Todo:

- [x] Rewrite text
- [x] Use `monospaced` font where needed
- [x] Convert Terrarform range constraints list into table

Source: https://github.com/renovatebot/renovate/blob/main/lib/modules/manager/terraform/readme.md

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Rewrite readme:
  - Use simple language
  - Add headings
  - Convert lists to tables
  - Use `monospace` styling for versions

## Context

Closes #17975.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/terraform/readme.md [hunk 1]
```diff
@@ -1,33 +1,80 @@
...
-- required_providers block ( Terraform >= 0.13.0)
-  - TerraformRegistry ( Public and Private )
-- required_version
-- helm_release
-  - chart repository ( Public and Private )
-- docker\_\*
-  - Docker registry ( Public and Private )
-- kubernetes\_\*
-  - Docker registry ( Public and Private )
-- [tfe_workspace](https://registry.terraform.io/providers/hashicorp/tfe/latest/docs/resources/workspace) ( `terraform_version` argument )
```
Comment (HonkingGoose) [inline lib/modules/manager/terraform/readme.md:18]: I'm converting this list into a table, as the current layout doesn't render properly on the published docs. The published docs have a "flat list" so you don't even see the sub-points.

I think chopping this up into new headings with a table is easier to read and understand.

File: lib/modules/manager/terraform/readme.md [hunk 2]
```diff
@@ -1,33 +1,80 @@
...
+Renovate can update `kubernetes\_\*` files.
+
+| Name            | Public hosting | Private hosting |
+| --------------- | :------------: | :-------------: |
+| Docker registry |      yes       |       yes       |
+
+#### tfe_workspaces
+
+Renovate can update [tfe_workspaces](https://registry.terraform.io/providers/hashicorp/tfe/latest/docs/resources/workspace).
+Renovate searches for the `terraform_version` argument.
```
Comment (HonkingGoose) [inline lib/modules/manager/terraform/readme.md:59]: Please double check me on the conversion from lists to table. 😉

File: lib/modules/manager/terraform/readme.md [hunk 3]
```diff
@@ -1,33 +1,80 @@
...
+| `<= 1.2.0`           | version 1.2.0 or older                                |
+| `~> 1.2.0`           | any non-beta version >= 1.2.0 and < 1.3.0, e.g. 1.2.X |
+| `~> 1.2`             | any non-beta version >= 1.2.0 and < 2.0.0, e.g. 1.X.Y |
+| `>= 1.0.0, <= 2.0.0` | any version between 1.0.0 and 2.0.0 inclusive         |
+
+### Disabling parts of the manager
+
+You can use these `depTypes` for fine-grained control, for example to disable parts of the Terraform manager.
+
+| resource                             |               `depType`                |                                   Notes                                    |
```
Comment (HonkingGoose) [inline lib/modules/manager/terraform/readme.md:77]:
```suggestion
| Resource                             |               `depType`                |                                   Notes                                    |
```

File: lib/modules/manager/terraform/readme.md [hunk 4]
```diff
@@ -1,33 +1,80 @@
...
+
+Renovate understands these Terraform range constraints:
+
+| Terraform range      | Notes                                                 |
+| -------------------- | ----------------------------------------------------- |
+| `>= 1.2.0`           | version 1.2.0 or newer                                |
+| `<= 1.2.0`           | version 1.2.0 or older                                |
+| `~> 1.2.0`           | any non-beta version >= 1.2.0 and < 1.3.0, e.g. 1.2.X |
+| `~> 1.2`             | any non-beta version >= 1.2.0 and < 2.0.0, e.g. 1.X.Y |
+| `>= 1.0.0, <= 2.0.0` | any version between 1.0.0 and 2.0.0 inclusive         |
```
Comment (HonkingGoose) [inline lib/modules/manager/terraform/readme.md:71]: This needs some `monospaced` styling as well:

```suggestion
| Terraform range      | Notes                                                       |
| -------------------- | ----------------------------------------------------------- |
| `>= 1.2.0`           | version `1.2.0` or newer                                    |
| `<= 1.2.0`           | version `1.2.0` or older                                    |
| `~> 1.2.0`           | any non-beta version `>= 1.2.0` and `< 1.3.0`, e.g. `1.2.X` |
| `~> 1.2`             | any non-beta version `>= 1.2.0` and `< 2.0.0`, e.g. `1.X.Y` |
| `>= 1.0.0, <= 2.0.0` | any version between `1.0.0` and `2.0.0` inclusive           |
```

File: lib/modules/manager/terraform/readme.md [hunk 5]
```diff
@@ -1,33 +1,80 @@
...
+
+Renovate supports updating the Terraform dependencies listed below.
+Check the tables to see where some dependencies can be hosted.
+
+#### Modules
+
+| Name              | Public hosting | Private hosting |
+| ----------------- | :------------: | :-------------: |
+| GitTags           |                |                 |
+| GithubTags        |                |                 |
```
Comment (secustor) [inline lib/modules/manager/terraform/readme.md:11]:
```suggestion
| GitTags           |        yes        |       yes          |
| GithubTags        |      yes          |       yes          |
```

File: lib/modules/manager/terraform/readme.md [hunk 6]
```diff
@@ -1,33 +1,80 @@
...
+
+Needs Terraform `>= 0.13.0`.
+
+| Name              | Public hosting | Private hosting |
+| ----------------- | :------------: | :-------------: |
+| TerraformRegistry |      yes       |       yes       |
+
+#### required_version
+
+Renovate can update `required_version` dependencies.
```
Comment (secustor) [inline lib/modules/manager/terraform/readme.md:32]:
```suggestion
Renovate can update the `required_version` attribute of the Terraform block.
```

File: lib/modules/manager/terraform/readme.md [hunk 7]
```diff
@@ -1,33 +1,80 @@
...
+
+#### helm_release
+
```
Comment (secustor) [inline lib/modules/manager/terraform/readme.md:35]:
```suggestion

Renovate can update the version attribute of  `helm_release` resources.

```

File: lib/modules/manager/terraform/readme.md [hunk 8]
```diff
@@ -1,33 +1,80 @@
...
+| Name             | Public hosting | Private hosting |
+| ---------------- | :------------: | :-------------: |
+| chart repository |      yes       |       yes       |
+
+#### Docker
+
+Renovate can update `docker\_\*` files.
```
Comment (secustor) [inline lib/modules/manager/terraform/readme.md:42]:
```suggestion
Renovate can update image references of the docker provider resources (`docker\_\*`).
```

File: lib/modules/manager/terraform/readme.md [hunk 9]
```diff
@@ -1,33 +1,80 @@
...
+
+| Name            | Public hosting | Private hosting |
+| --------------- | :------------: | :-------------: |
+| Docker registry |      yes       |       yes       |
+
+#### Kubernetes
+
+Renovate can update `kubernetes\_\*` files.
```
Comment (secustor) [inline lib/modules/manager/terraform/readme.md:50]:
```suggestion
Renovate can update image references of Kubernetes provider resources (`kubernetes\_\*`).
```

**Review by secustor — COMMENTED** [review]
Some suggestions which use Terraform terminology .

**Review by secustor — APPROVED** [review]

**Pull request merged**